### PR TITLE
OUT-653, OUT-650 Filter by assignee' is shown in mobile view even if IU selects 'My tasks' filter

### PR DESCRIPTION
**Pull request by arpandhakal**
- filter by assignee removed when my tasks is clicked in filter button options

### Tasks

- [Filter by assignee' is shown in mobile view even if IU selects 'My tasks' filter](https://linear.app/copilotplatforms/issue/OUT-653/filter-by-assignee-is-shown-in-mobile-view-even-if-iu-selects-my-tasks)
- [Filter by assignee' drop-down shows the 'No assignee' option in mobile experience. but it is not shown on the desktop view.](https://linear.app/copilotplatforms/issue/OUT-650/filter-by-assignee-drop-down-shows-no-assignee-option-on-mobile-but)

### Changes

- [x] no assignee option removed on mobile view on filter by assignee
- [x] filter by assignee removed when my tasks is clicked in filter button options 

### Testing Criteria

- [LOOM](https://www.loom.com/share/d2414f943c804c568e9f7c7b1fe1a19a)

### CODE REVIEW

**Comment by rrojan** [comment]
Design and functionality wise everything looks good, however please make these code quality changes @arpandhakal

File: src/components/cards/ListViewTaskCard.tsx [hunk 1]
```diff
@@ -51,122 +51,118 @@ export const ListViewTaskCard = ({
...
                   variant="sm"
-                  fontWeight={400}
                   sx={{
-                    color: (theme) => theme.color.gray[500],
-                    flexGrow: 0,
-                    minWidth: '60px',
                     lineHeight: '21px',
+                    wordBreak: 'break-word',
+                    flexGrow: 1,
+                    minWidth: '0',
```
[the remark this answers is not in the document]
Comment (rrojan) [reply src/components/cards/ListViewTaskCard.tsx:75]: Is the `minWidth: 0` supposed to allow us to shrink the component if necessary?

File: src/components/cards/ListViewTaskCard.tsx [hunk 2]
```diff
@@ -102,28 +102,18 @@ export const ListViewTaskCard = ({
                 getSelectedValue={(_newValue) => {
                   const newValue = _newValue as IAssigneeCombined
                   updateAssigneeValue(newValue)
-                  const assigneeType = newValue?.type
+                  const assigneeType = newValue.type as AssigneeType
```
Comment (rrojan) [inline src/components/cards/ListViewTaskCard.tsx:105]: Are we sure that there is no case where `newValue` can be undefined / null here?

Comment (arpandhakal) [reply src/components/cards/ListViewTaskCard.tsx:105]: in IAssigneeCombined :  type: AssigneeType, I think type can never be undefined for an assignee value.

Comment (arpandhakal) [reply src/components/cards/ListViewTaskCard.tsx:105]: On second thoughts, there should be no assignee whose value will be null in the future, so I think it should be handled. Thanks for pointing that out

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Pull request merged**